source/features/pull-request-timeline-filter.tsx

```diff
@@ -0,0 +1,260 @@
+import React from 'dom-chef';
+import select from 'select-dom';
+import elementReady from 'element-ready';
+import {observe} from 'selector-observer';
+import * as pageDetect from 'github-url-detection';
+
+import features from '.';
+
+async function sleep(seconds: number): Promise<any> {
+	await new Promise(resolve => setTimeout(resolve, seconds * 1000));
+}
+
+interface FilterSettings {
+	HideUnresolved: boolean;
+	HideResolved: boolean;
+	hideNormalComment: boolean;
+	HideCommits: boolean;
+	AutoLoadHidden: boolean;
+	HideOthers: boolean;
+}
+
+const CurrentSettings: FilterSettings =
+{
+	HideResolved: true,
+	HideUnresolved: false,
+	hideNormalComment: false,
+	HideOthers: false,
+	HideCommits: false,
+	AutoLoadHidden: false
+};
+
+const hideUnresolvedSelectorId = 'hide-unresolved';
+const hideResolvedSelectorId = 'hide-resolved';
+const hideNormalCommentsSelectorId = 'hide-normal-comments';
+const hideOthersSelectorId = 'hide-others';
+const hideCommitsSelectorId = 'hide-commits';
+const autoLoadHiddenSelectorId = 'auto-load-hidden';
+
+const timelineFiltersSelectorId = 'timeline-filters';
+const detailsSelector = `#${timelineFiltersSelectorId} details`;
+const notiticationsSelector = '.discussion-sidebar-item.sidebar-notifications';
+const timelineItemSelector = '.js-timeline-item';
+const loadMoreSelector = '.ajax-pagination-btn:not([disabled])';
+
+function regenerateFilterSummary(): void {
+	const timelineFilter = select(`#${timelineFiltersSelectorId}`)!;
+	const newSummary = (
+		<p className="reason text-small text-gray">
+			{CurrentSettings.HideUnresolved ? 'Hide' : 'Show'} unresolved comments. <br/>
+			{CurrentSettings.HideResolved ? 'Hide' : 'Show'} resolved comments. <br/>
+			{CurrentSettings.hideNormalComment ? 'Hide' : 'Show'} normal comments. <br/>
+			{CurrentSettings.HideOthers ? 'Hide' : 'Show'} other items. <br/>
+			{CurrentSettings.HideCommits ? 'Hide' : 'Show'} commits. <br/>
```

Comment:
Like for labels, this section should only show non-default options, that is: only show these lines if they are “Hide”

Reply:
Done